python/grass/jupyter/tests/timeseries_test.py

```diff
@@ -0,0 +1,82 @@
+"""Test TimeSeries functions"""
+
+
+from pathlib import Path
+import pytest
+
+try:
+    import IPython
+except ImportError:
+    IPython = None
+
+try:
+    import ipywidgets
+except ImportError:
+    ipywidgets = None
+
+import grass.jupyter as gj
+
+
+def test_fill_none_values():
+    """Test that fill_none_values replaces None with previous value in list"""
+    names = ["r1", "None", "r3"]
+    fill_names = gj.fill_none_values(names)
+    assert fill_names == ["r1", "r1", "r3"]
+
+
+def test_collect_layers(space_time_raster_dataset):
+    """Check that collect layers returns list of layers and dates"""
+    names, dates = gj.collect_layers(
+        space_time_raster_dataset.name, fill_gaps=False, element_type="strds"
+    )
+    # Test fill_gaps=False at empty time step
+    assert names[1] == "None"
+    assert dates[1] == "2001-02-01 00:00:00"
+    # Remove the empty time step - see space_time_raster_dataset creation
+    names.pop(1)
+    dates.pop(1)
+    assert names == space_time_raster_dataset.raster_names
+    assert len(dates) == len(space_time_raster_dataset.start_times)
+    assert len(names) == len(dates)
+
+
+def test_method_call_collector():
+    """Check that MethodCallCollector constructs and collects GRASS calls"""
+    mcc = gj.MethodCallCollector()
+    mcc.d_rast(map="elevation")
+    module, kwargs = mcc.calls[0]
+    assert module == "d.rast"
+    assert kwargs["map"] == "elevation"
+
+
+def test_default_init(space_time_raster_dataset):
+    """Check that TimeSeries init runs with default parameters"""
+    img = gj.TimeSeries(space_time_raster_dataset.name)
+    assert img.timeseries == space_time_raster_dataset.name
+
+
+@pytest.mark.parametrize("fill_gaps", ["False", "True"])
```

Comment:
I'm not sure what you mean - `test_renderer_layers` runs twice, once with `fill_gaps=True` and again with `fill_gaps=False`. The `test_fill_none_values` tests replacing `None` values with the previous values and `test_collect_layers` uses `fill_gaps=False` too.

Reply:
> I'm not sure what you mean - `test_renderer_layers` runs twice, once with `fill_gaps=True` and again with `fill_gaps=False`.

The issue is that it runs with `fill_gaps="True"` and again with `fill_gaps="False"` so `bool(fill_gaps)` is always True. You need to remove the quotes.

```python
>>> bool("False")  # This is what you have there now.
True
>>> bool(False)
False
>>> False  # This is all you need here.
False
>>> import ast
>>> ast.literal_eval("False")
False
```

> The `test_fill_none_values` tests replacing `None` values with the previous values and `test_collect_layers` uses `fill_gaps=False` too.

Thanks. I didn't notice test_collect_layers=False, so that part is resolved for me now. We have a more focused test with `fill_gaps=False`.

Reply:
Ah! Ok, I see now. I removed the quotes.